lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/ext/UmAppDatabaseMigrations.kt

```diff
@@ -1653,6 +1653,43 @@ val MIGRATION_201_202 = DoorMigrationStatementList(201, 202) { db ->
     }
 }
 
+val MIGRATION_202_203_SERVER = DoorMigrationStatementList(202, 203) { db ->
```

Comment:
The migration needs to use the ValidateUsernameUseCase so it follows the don't repeat yourself principle -e.g. the logic for what is, and is not, a valid username, should be in one place.

If you put this migration into ```jvmMain``` source set, you can then use ```db.connection``` to get the underlying JDBC database connection so you can run any JDBC query or statement.

It should go through all usernames in batches, check the username, and then if a username is found where the return result of validateusernameusecase is not the same as the username (e.g. something in the username is invalid), it should be updated (e.g. the update statement should only run on the usernames that need changed).

The migration as-below has different logic. It would not find white space other than space (e.g. won't find tab, carriage return, etc). It also won't find or fix usernames with special characters.

Reply:
Note: you may need to put the server side migration in app-ktor-server or core/jvmMain source set, otherwise, it won't have access to the ValidateUsernameUseCase.